src/processors/TypeScriptInputProcessor.ts

```diff
@@ -0,0 +1,63 @@
+import { CommonInputModel } from '../models';
+import { resolve } from 'path';
+import ts from 'typescript';
+import * as fs from 'fs';
+import * as TJS from 'typescript-json-schema';
+import { JsonSchemaInputProcessor } from './JsonSchemaInputProcessor';
+import { AbstractInputProcessor } from './AbstractInputProcessor';
+
+/** Class for processing Typescript code inputs to Common module*/
+export class TypeScriptInputProcessor extends AbstractInputProcessor {
+  static settings: TJS.PartialArgs = {
+    uniqueNames: true,
+    required: true
+  }; 
+
+  static compilerOptions: TJS.CompilerOptions = {
+    strictNullChecks: true
+  };
+
+  static generateProgram(basePath: string, requiredFile: string): ts.Program {
+    return TJS.getProgramFromFiles([resolve(requiredFile)], TypeScriptInputProcessor.compilerOptions, basePath);
+  }
+
+  static generateJSONSchema(basePath: string, pathToFile: string, typeRequired: string): TJS.Definition | null {
+    const program: ts.Program = TypeScriptInputProcessor.generateProgram(basePath, pathToFile);
+
+    // if (multipleFiles) {
+    //   const generator = TJS.buildGenerator(program, TypeScriptInputProcessor.settings);
+    //   if (generator) {
+    //     return TJS.generateSchema(program, typeRequired, TypeScriptInputProcessor.settings, [], generator);
+    //   }
+    // }
+  
+    return TJS.generateSchema(program, typeRequired, TypeScriptInputProcessor.settings);
+  }
+
+  shouldProcess(input: Record<string, any>): boolean {
+    if (typeof input !== 'object' || typeof input.basePath !== 'string') {return false;}
+    return true;
+  }
+
+  process(input: Record<string, any>): Promise<CommonInputModel> {
```

Comment:
You can use `input: any` here 🙂 We will just have to change the abstract processor as well 🙂